### DISCUSSION TITLE
Welcome to DfRefactor Discussions!

### DISCUSSION THREAD

**Discussion by NilsSve** [NilsSve/DfRefactor#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mwpmullan** [comment]
In the EditorView.vw, on line 27 we have "Delegate Set phoEditor to (Self)" unfortunately this throws an error at runtime (Unresolved object reference), I'm not sure what you were aiming for here, my suggested refactor would be to move the property to the desktop, and only have one defined, rather than several, but you may have reasons that preclude this.

**Comment by NilsSve** [comment]
Thanks for the report! I'll look into it.

The EditorView.vw is part of the DFRefactor.src program and we haven't got to that part yet.
Therefore, during the Alpha-testing phase, it would be appreciated if we all could focus on the TestBench (and optionally the DFUnit_TestRunner) program.

**Comment by mwpmullan** [comment]
I’m doing a paid project which includes a need for conv2000 to be rewritten. I figured I’d add it as a tool to dfrefactor. It seems like a good fit.

For now I’m making it a separate tool, because it didn’t quite fit exactly into DFRefactor.src.



[X]

Michael Mullan


PTI - PACKAGING TECHNOLOGIES & INSPECTION

8 Skyline Drive  |   Hawthorne, NY 10532   USA  |   Office:  914.337.2005  |   Cell:  917.299.7983

Contact Us<http://www.ptiusa.com/About-PTI/Contact-Us>   |    www.ptiusa.com<http://www.ptiusa.com/>   |   www.ptipacktech.com<http://www.ptipacktech.com/>   |  Upcoming events for PTI<https://www.ptiusa.com/About-PTI/Trade-Shows-Conferences>    |   follow us on LinkedIn


On Nov 5, 2023, at 12:21 PM, Nils Svedmyr ***@***.***> wrote:

﻿

Thanks for the report! I'll look into it.

The EditorView.vw is part of the DFRefactor.src program and we haven't got to that part yet.
Therefore, during the Alpha-testing phase, it would be appreciated if we all could focus on the TestBench (and optionally the DFUnit_TestRunner) program.

**Comment by NilsSve** [comment]
I had a quick look at the line in EditorView.vw:
`    Delegate Set phoEditor to (Self)`

And I believe it is correct. It sets the phoEditor property of the oEditorView_vw which is a cRefactorDbView class were it is defined.

When testing I selected a workspace, then from the toolbar selected "Select File", selected a single refactoring function, and hit the "Start Refactoring" button. 
I could not get that line to fail. So I need more info on what you tried.

**Comment by mwpmullan** [comment]
I just compiled and ran the 64 bit version of the program.  The error appears at startup.

A fresh clone from the repository today.
________________________________

**Comment by NilsSve** [comment]
It shouldn't make any difference for the phoEditor property if compiled under x86 or x64, so there must be something else that gets in the way.

There have been several new commits and pushes today, so perhaps you don't have the latest after all?
As explained in the  "Alpha test info DFrefactor project.txt" document, all Alpha testing should be done in 32-bit.

Could you please email me the Conv2000 source so I can have a look at it?

**Comment by mwpmullan** [comment]
I will send you the source when it's finished. :-)

**Comment by NilsSve** [comment]
Great, thank you!
I will then do my best to include it in the framework.

**Comment by NilsSve** [comment]
This thread has been closed. Open a new one if needed.
